src/battle_util.c

```diff
@@ -6619,7 +6638,8 @@ bool32 CanBeSlept(u32 battler, u32 ability)
      || gBattleMons[battler].status1 & STATUS1_ANY
      || IsAbilityOnSide(battler, ABILITY_SWEET_VEIL)
      || IsAbilityStatusProtected(battler)
-     || IsBattlerTerrainAffected(battler, STATUS_FIELD_ELECTRIC_TERRAIN | STATUS_FIELD_MISTY_TERRAIN))
+     || IsBattlerTerrainAffected(battler, STATUS_FIELD_ELECTRIC_TERRAIN | STATUS_FIELD_MISTY_TERRAIN)
+     || (isBlockedBySleepClause && IsSleepClauseActiveForSide(GetBattlerSide(battler))))
```

Comment:
I would make a separate if-statement in this function above this one that returns early just for the sleep clause and additionally check the config as the first thing.

Reply:
did this, but also the first thing inside `IsSleepClauseActiveForSide` is a check for  `FlagGet(B_FLAG_SLEEP_CLAUSE)` so i didnt add an additional check outside it

Reply:
I see. That makes sense